gobblin-core/src/main/java/org/apache/gobblin/async/AsyncRequest.java

```diff
@@ -61,7 +61,7 @@ public long getBytesWritten() {
   /**
    * Get all records information in the request
    */
-  public List<Thunk<D>> getThunks() {
+  public List<Thunk<D>> getThunksCopy() {
```

Comment:
Avoid back incompatible change. There are 3rd party codes already using this method.

Reply:
Sure. One follow up question here is why do we have a copy instead of returning a reference to thunk container. Any specific use case for that?

Reply:
Talked offline, copy was meant to protect `AsyncRequest` from unexpected modification.